cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/common/BatchFileFilter.java

```diff
@@ -108,11 +113,32 @@ public void setConf(Configuration conf) {
       regex = Pattern.compile(input);
     }
     lastRead = conf.get(FileBatchSource.LAST_TIME_READ, "-1");
+
+    if (!lastRead.equals("-1")) {
+      dateRangesToRead = GSON.fromJson(lastRead, ARRAYLIST_DATE_TYPE);
+    }
+
     try {
       prevMinute = sdf.parse(conf.get(FileBatchSource.CUTOFF_READ_TIME));
     } catch (ParseException pe) {
       prevMinute = new Date(System.currentTimeMillis());
     }
   }
-}
 
+  /**
+   * Determines if a file should be read in
+   *
+   * @param fileDate when the file was created
+   * @return true if the file is to be read, false otherwise
+   */
+  private boolean toRead(Date fileDate) {
+    for (int i = 0; i < dateRangesToRead.size() / 2; i++) {
+      if (fileDate.compareTo(dateRangesToRead.get(2 * i)) > 0 &&
```

Comment:
Seems like the time ranges are exclusive of the begin value, and inclusive of the end value?